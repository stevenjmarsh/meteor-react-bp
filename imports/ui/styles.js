import { css } from "glamor";

/**
 *  Component Style Pattern (use with glamor)
 *
 *  const componentRules = {
 *    rule1: "value",
 *      ...
 *    ruleN: "value",
 *  };
 *  export const componentStyle = {
 *    rules: componentRules,
 *    className: `${css(componentRules)}`,
 *  };
 *
 */

const gradientBackgroundRules = {
  background: [
    "#048ec5",
    "-moz-linear-gradient(-45deg, #048ec5 0%, #2ecc71 100%) fixed",
    "-webkit-linear-gradient(-45deg, #048ec5 0%,#2ecc71 100%) fixed",
    "linear-gradient(135deg, #048ec5 0%,#2ecc71 100%) fixed",
  ],
  /* IE6-9 fallback on horizontal gradient */
  filter: "progid:DXImageTransform.Microsoft.gradient( startColorstr='#048ec5', endColorstr='#2ecc71',GradientType=1 )",
};

const defaultPageRules = {
  ...gradientBackgroundRules,
  color: "white",
  padding: "30px",
};
export const defaultPageStyle = { // eslint-disable-line import/prefer-default-export
  rules: defaultPageRules,
  className: `${css(defaultPageRules)}`,
};
