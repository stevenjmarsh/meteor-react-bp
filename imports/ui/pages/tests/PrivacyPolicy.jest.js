/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */

import React from "react";
import renderer from "react-test-renderer";
import PrivacyPolicy from "../PrivacyPolicy";
import {
  formattedJSON,
  getAllComponentStyle,
} from "../../../utils/tests/testUtils";
import * as styles from "../../../ui/styles";

describe("<PrivacyPolicy />", function () {
  it("matches render snapshot", function () {
    const tree = renderer.create(<PrivacyPolicy />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("matches style / css snapshot", function () {
    const allStyle = getAllComponentStyle(<PrivacyPolicy />, styles);
    expect(formattedJSON(allStyle)).toMatchSnapshot();
  });
});
