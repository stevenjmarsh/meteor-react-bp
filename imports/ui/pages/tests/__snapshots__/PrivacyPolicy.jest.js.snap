exports[`<PrivacyPolicy /> matches render snapshot 1`] = `
<div
  className="css-wlbfil">
  <h1>
    Privacy Policy
  </h1>
  <p>
    This privacy policy discloses the privacy practices for Eat Drink Healthy. This privacy policy applies solely to information collected by this web application and site. It will notify you of the following:
  </p>
  <ol>
    <li>
      What personally identifiable information is collected from you through the web site or app, how it is used and with whom it may be shared.
    </li>
    <li>
      What choices are available to you regarding the use of your data.
    </li>
    <li>
      The security procedures in place to protect the misuse of your information.
    </li>
    <li>
      How you can correct any inaccuracies in the information.
    </li>
  </ol>
  <h2>
    Information Collection, Use, and Sharing 
  </h2>
  <p>
    We are the sole owners of the information collected on this site. We only have access to/collect information that you voluntarily give us via registration or other direct contact from you. We will not sell or rent this information to anyone.
  </p>
  <p>
    We will use your information to personalize your experience with the website and application, or respond to you regarding the reason you contacted us. We will not share your information with any third party outside of our organization.
  </p>
  <p>
    Unless you ask us not to, we may contact you via email in the future to tell you about product updates, or changes to this privacy policy.
  </p>
  <h2>
    Your Access to and Control Over Information
  </h2>
  <p>
    You may opt out of any future contacts from us at any time. You can do the following at any time by contacting us via the contact form on our support page:
  </p>
  <ul>
    <li>
      See what data we have about you, if any.
    </li>
    <li>
      Change/correct any data we have about you.
    </li>
    <li>
      Have us delete any data we have about you.
    </li>
    <li>
      Express any concern you have about our use of your data.
    </li>
  </ul>
  <h2>
    Security
  </h2>
  <p>
    We take precautions to protect your information. When you submit sensitive information via the website or app, your information is protected both online and offline.
  </p>
  <p>
    Wherever we collect sensitive information (such as credit card data), that information is encrypted and transmitted to us in a secure way. You can verify this by looking for a closed lock icon at the bottom of your web browser, or looking for \"https\" at the beginning of the address of the web page.
  </p>
  <p>
    While we use encryption to protect sensitive information transmitted online, we also protect your information offline. Only employees who need the information to perform a specific job (for example, billing or customer service) are granted access to personally identifiable information. The computers/servers in which we store personally identifiable information are kept in a secure environment.
  </p>
  <h2>
    Links
  </h2>
  <p>
    This web site and app may contain links to other sites. Please be aware that we are not responsible for the content or privacy practices of such other sites. We encourage our users to be aware when they leave our site or app and to read the privacy statements of any other site that collects personally identifiable information.
  </p>
  <h2>
    Updates
  </h2>
  <p>
    Our Privacy Policy may change from time to time and all updates will be posted on this page.
  </p>
  <p>
    If you feel that we are not abiding by this privacy policy, you should contact us immediately via the contact form on our support page.
  </p>
</div>
`;

exports[`<PrivacyPolicy /> matches style / css snapshot 1`] = `
"[
  {
    \"rules\": {
      \"background\": [
        \"#048ec5\",
        \"-moz-linear-gradient(-45deg, #048ec5 0%, #2ecc71 100%) fixed\",
        \"-webkit-linear-gradient(-45deg, #048ec5 0%,#2ecc71 100%) fixed\",
        \"-webkit-linear-gradient(135deg, #048ec5 0%,#2ecc71 100%) fixed\",
        \"-moz-linear-gradient(135deg, #048ec5 0%,#2ecc71 100%) fixed\",
        \"linear-gradient(135deg, #048ec5 0%,#2ecc71 100%) fixed\"
      ],
      \"filter\": \"progid:DXImageTransform.Microsoft.gradient( startColorstr=\'#048ec5\', endColorstr=\'#2ecc71\',GradientType=1 )\",
      \"color\": \"white\",
      \"padding\": \"30px\"
    },
    \"className\": \"css-wlbfil\"
  }
]"
`;
